cmd/healthchecker/options/options.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"flag"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	"k8s.io/node-problem-detector/pkg/healthchecker/types"
+)
+
+// NewHealthCheckerOptions returns an empty health check options struct.
+func NewHealthCheckerOptions() *HealthCheckerOptions {
+	return &HealthCheckerOptions{}
+}
+
+// HealthCheckerOptions are the options used to configure the health checker.
+type HealthCheckerOptions struct {
+	Component          string
+	ContainerRuntime   string
+	EnableRepair       bool
+	CriCtlPath         string
+	CoolDownTime       time.Duration
+	HealthCheckTimeout time.Duration
+}
+
+// AddFlags adds health checker command line options to pflag.
+func (hco *HealthCheckerOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&hco.Component, "component", types.KubeletComponent,
+		"The component to check health for. Supports kubelet and container-runtime.")
+	fs.StringVar(&hco.ContainerRuntime, "container-runtime", types.DockerRuntime,
+		"The underlying container runtime name on the node. This is ignored if component is not container-runtime. Supports docker and containerd.")
+	fs.BoolVar(&hco.EnableRepair, "enable-repair", true, "Flag to enable/disable repair attempt for the component.")
+	fs.StringVar(&hco.CriCtlPath, "crictl-path", types.DefaultCriCtl,
+		"The path to the crictl binary. This is used to check health of container-runtimes other than docker.")
+	fs.DurationVar(&hco.CoolDownTime, "cooldown-time", types.DefaultCoolDownTime,
+		"The time to wait after repairing the component. This helps in maintaining a buffer between repair and next invocation.")
+	fs.DurationVar(&hco.HealthCheckTimeout, "health-check-timeout", types.DefaultHealthCheckTimeout,
+		"The time to wait before marking the component as unhealthy.")
+}
+
+// ValidOrDie validates health checker command line options.
+func (hco *HealthCheckerOptions) ValidOrDie() {
+	// Make sure the component specified is valid.
+	if hco.Component != types.KubeletComponent && hco.Component != types.ContainerRuntimeComponent {
+		panic("The component specified is not supported. Supported components are : <docker/container-runtime>")
```

Comment:
Supported components don't have `kubelet`.

Reply:
Was a typo there. Updated the flags.